<launch>
  <arg name="PID_IIR_LPF_bandwidth" default="17" />
  <arg name="imu_filter_rate" default="100"/> <!-- Leave at 100 Hz -->

  <rosparam command="load" ns="roll_controller" file="$(find riptide_controllers)/cfg/pid_config_roll.yaml" />
  <rosparam command="load" ns="pitch_controller" file="$(find riptide_controllers)/cfg/pid_config_pitch.yaml" />
  <rosparam command="load" ns="yaw_controller" file="$(find riptide_controllers)/cfg/pid_config_yaw.yaml" />

  <node pkg="riptide_controllers" type="attitude_controller" name="attitude_controller" output="screen" >
    <rosparam file="$(find riptide_controllers)/cfg/controller_properties_attitude.yaml" command="load" />
    <param name="PID_IIR_LPF_bandwidth" type="double" value="$(arg PID_IIR_LPF_bandwidth)" />
    <param name="imu_filter_rate" type="double" value="$(arg imu_filter_rate)" />
  </node>
</launch>
